packages/amplify-provider-awscloudformation/src/aws-utils/IdentityPoolService.ts

```diff
@@ -102,9 +102,8 @@ export class IdentityPoolService implements IIdentityPoolService {
 
     if (arn) {
       const parts = arn.split('/');
-
-      if (parts.length === 2) {
-        resourceName = parts[1];
+      if (parts?.length >= 2) {
+        resourceName = parts.at(-1);
       }
```

Comment:
could we replace this logic with https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_util_arn_parser.html ?

Reply:
Great suggestion! I'll create a wrapper function in `amplify-cli-core` so that the dependency on `@aws-sdk/utility-arn-parser` v3 can be isolated to one package for now. Thoughts?

Reply:
not sure if that's necessary. if we `^` version of that yarn should take care of duplicates. 
first check if this is solving our problem...